chapters/ch06/2.related-keywords-from-signals.ipynb

```diff
@@ -340,21 +339,21 @@
   },
   {
    "cell_type": "code",
-   "execution_count": 21,
+   "execution_count": 11,
    "metadata": {},
    "outputs": [
     {
      "name": "stdout",
      "output_type": "stream",
      "text": [
       "Original signals format: \n",
-      "+--------------------+-----------+--------------------+--------+-----+-------+\n",
-      "|                  id|   query_id|         signal_time|  target| type|   user|\n",
-      "+--------------------+-----------+--------------------+--------+-----+-------+\n",
-      "|00000571-ce8b-4ed...| u93128_0_1|2019-10-14 23:12:...| Fallout|query| u93128|\n",
-      "|00000d9b-c06a-466...|u513170_0_1|2019-10-22 17:25:...|Seinfeld|query|u513170|\n",
-      "|000014f4-d8c6-4c1...| u87599_0_1|2020-02-23 22:00:...| laptops|query| u87599|\n",
-      "+--------------------+-----------+--------------------+--------+-----+-------+\n",
```

Comment:
Alright, after some investigation, I've identified a side effect of the inconsistent "id field" policies between the search engines. This issue will cause null id's within spark view created from an opensearch collection because opensearch uses the "_id" field. This is a bug for any sql query that uses the id field of a spark view created from an opensearch collection, which is not a currently used use case in the code. 

I'm kicking this can down the road as it doesn't effect any code in the codebase. I'll discuss this further in the meeting.

I've specified the column's to print within the print function, and re-run this as solr.

Reply:
A fix has been created by parsing the metadata field being returned by spark loader for opensearch